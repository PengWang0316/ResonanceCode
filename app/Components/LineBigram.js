import React from "react";
import PropTypes from "prop-types";

const LineBigram = (props)=>{
  // let image= ;
  // console.log(props.data.image);
  let images = {
    "../imgs/conception.png":require("../imgs/conception.png"),
    "../imgs/growth.png":require("../imgs/growth.png"),
    "../imgs/maturation.png":require("../imgs/maturation.png"),
    "../imgs/seeding.png":require("../imgs/seeding.png")
  };
  return(
    <div className="lineDesDiv">
      <div className="rcTitle">{props.title} <img src={props.data.image?images[props.data.image]:""} alt={props.data.name} /> </div>

      {props.data.name && <div>Name: {props.data.name}</div>}

      <div>{props.data.energy_state && <span>Energy State: {props.data.energy_state}</span>}{props.data.manifestation && <span>Manifestation: {props.data.manifestation}</span>}{props.data.possibilities && <span>Possibilities: {props.data.possibilities}</span>}</div>

      <div>Question:</div>
      <div>{props.data.question}</div>

    </div>
  );
};
LineBigram.proptypes={
  title: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired
};
export default LineBigram;
