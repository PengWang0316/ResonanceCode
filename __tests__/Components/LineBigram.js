import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import LineBigram from '../../app/Components/LineBigram';

describe('LineBigram test', () => {
  const defaultProps = {
    title: 'title',
    data: {
      image: '../imgs/conception.png', name: 'name', energy_state: 'energyState', manifestation: 'manifestation', possibilities: 'possibilities', question: 'question'
    }
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<LineBigram {...props} />);

  test('LineBigram has image snapshot', () => expect(renderer.create(<LineBigram {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('LineBigram has no image snapshot', () => expect(renderer.create(<LineBigram {...{
 ...defaultProps,
data: {
    name: 'name', energy_state: 'energyState', manifestation: 'manifestation', possibilities: 'possibilities', question: 'question'
  }
}}
  />).toJSON()).toMatchSnapshot());
});
