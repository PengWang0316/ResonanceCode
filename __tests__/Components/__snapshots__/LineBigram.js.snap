// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineBigram test LineBigram has image snapshot 1`] = `
<div
  className="mt-2"
>
  <div
    className="font-weight-bold"
  >
    title
     
    <img
      alt="name"
      src={
        Object {
          "process": [Function],
        }
      }
    />
     
  </div>
  <div>
    <b>
      Name:
    </b>
     
    name
  </div>
  <div
    className="font-weight-bold"
  >
    Questions:
  </div>
  <div />
</div>
`;

exports[`LineBigram test LineBigram has no image snapshot 1`] = `
<div
  className="mt-2"
>
  <div
    className="font-weight-bold"
  >
    title
     
    <img
      alt="name"
      src=""
    />
     
  </div>
  <div>
    <b>
      Name:
    </b>
     
    name
  </div>
  <div
    className="font-weight-bold"
  >
    Questions:
  </div>
  <div />
</div>
`;
